### PR TITLE
Update mlmd and pipelines service to use constructUrl for URL generation

### PR DESCRIPTION
**Pull request by Gkrumbach07**
https://issues.redhat.com/browse/RHOAIENG-12000

## Description

adds a constructUrl option that can be used instead of manually entering service info for backend proxy helper
uses this for mlmd and pipelines

## How Has This Been Tested?

make sure a pipelines server and mlmd server can fetch data
create a new pipeline server and verify backend is not crashing

## Test Impact

none backend 

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: backend/src/routes/api/service/mlmd/index.ts [hunk 1]
```diff
@@ -9,8 +9,7 @@ export default proxyService<DSPipelineKind>(
     plural: 'datasciencepipelinesapplications',
   },
   {
-    port: 8443,
-    prefix: 'ds-pipeline-md-',
+    constructUrl: (resource) => resource.status.components.mlmdProxy.url,
```
Comment (DaoDaoNoCode) [inline backend/src/routes/api/service/mlmd/index.ts:12]: Do you need to check status?
```suggestion
    constructUrl: (resource) => resource.status?.components.mlmdProxy.url,
```

File: backend/src/routes/api/service/pipelines/index.ts [hunk 2]
```diff
@@ -9,8 +9,7 @@ export default proxyService<DSPipelineKind>(
     plural: 'datasciencepipelinesapplications',
   },
   {
-    port: 8443,
-    prefix: 'ds-pipeline-',
+    constructUrl: (resource) => resource.status.components.apiServer.url,
```
Comment (DaoDaoNoCode) [inline backend/src/routes/api/service/pipelines/index.ts:12]: Same here

File: backend/src/routes/api/service/mlmd/index.ts (hunk 1 above)
Comment (Gkrumbach07) [reply backend/src/routes/api/service/mlmd/index.ts:12]: yeah i think so. i was wondering why TS was not yelling at me for that though

Comment (DaoDaoNoCode) [reply backend/src/routes/api/service/mlmd/index.ts:12]: The backend and frontend folders use different TS configs, and the backend one is much more tolerant.

Comment (Gkrumbach07) [reply backend/src/routes/api/service/mlmd/index.ts]: shoudl be fixed now

**Review by jpuzz0 — APPROVED** [review]
Verified [here](https://console-openshift-console.apps.kserve-pm.dev.datahub.redhat.com/) by creating a pipeline server and verifying mlmd resources were fetched after creating a new run with scalar metrics.

**Pull request merged**